packages/parser-package-json/src/parser.ts

```diff
@@ -0,0 +1,77 @@
+import * as path from 'path';
+
+import { FetchEnd, IJSONResult, Parser, SchemaValidationResult } from 'hint/dist/src/lib/types';
+import { Engine } from 'hint';
+import loadJSONFile from 'hint/dist/src/lib/utils/fs/load-json-file';
+import { parseJSON } from 'hint/dist/src/lib/utils/json-parser';
+import { validate } from 'hint/dist/src/lib/utils/schema-validator';
+
+import { PackageJsonEvents } from './types';
+
+export * from './types';
+
+export default class PackageJsonParser extends Parser<PackageJsonEvents> {
+    private schema: any;
+
+    public constructor(engine: Engine<PackageJsonEvents>) {
+        super(engine, 'package-json');
+        this.schema = loadJSONFile(path.join(__dirname, 'schema.json'));
+
+        /**
+         * package.json => type: 'json' (file type from extention).
+         */
+        engine.on('fetch::end::json', this.parsePackageJson.bind(this));
+    }
+
+    private validateSchema(config: any, resource: string, result: IJSONResult): SchemaValidationResult {
+        return validate(this.schema, config, result.getLocation);
+    }
+
+    private async emitInvalidPackageJson (validationResult: SchemaValidationResult, resource: string): Promise<void> {
+        await this.engine.emitAsync('parse::error::package-json::schema', {
+            error: new Error('Invalid package.json configuration'),
+            errors: validationResult.errors,
+            prettifiedErrors: validationResult.prettifiedErrors,
+            resource
+        });
+    }
+
+    private async parsePackageJson(fetchEnd: FetchEnd) {
+        const resource = fetchEnd.resource;
+        const resourceFileName = path.basename(resource);
+        const isPackageJson: boolean = resourceFileName === 'package.json';
+
+        if (!isPackageJson) {
+            return;
+        }
+
+        try {
+            const response = fetchEnd.response;
+            // When using local connector to read local files, 'content' is empty.
```

Comment:
Is this comment left-over from something? I don't think it's true and I don't see any code that attempts to handle empty content either.